compiler/rustc_middle/src/ty/crate_variances_map.rs

```diff
@@ -0,0 +1,17 @@
+use rustc_hir::def_id::DefIdMap;
+
+use crate::ty;
+
+/// The crate variances map is computed during typeck and contains the
+/// variance of every item in the local crate. You should not use it
+/// directly, because to do so will make your pass dependent on the
+/// HIR of every item in the local crate. Instead, use
+/// `tcx.variances_of()` to get the variance for a *particular*
+/// item.
+#[derive(HashStable, Debug)]
+pub struct CrateVariancesMap<'tcx> {
```

Comment:
I wonder if we should put these one-off query return values somewhere better instead of one module per item.

Reply:
Maybe they should just be put into `rustc_middle::ty::query` or something...

Reply:
Please don't. `ty::query` is there to isolate the macro-based plumbing.